ui/src/treeDataProvider/runWithLoadingNotification.ts

```diff
@@ -3,20 +3,31 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { CancellationToken, ProgressLocation, window } from 'vscode';
+import { CancellationToken, EventEmitter, ProgressLocation, window } from 'vscode';
 import * as types from '../../index';
 import { localize } from '../localize';
 
+let notificationActive: boolean = false;
+
 export async function runWithLoadingNotification<T>(context: types.ILoadingTreeContext, callback: (cancellationToken: CancellationToken) => Promise<T>): Promise<T> {
-    return await window.withProgress({ location: ProgressLocation.Notification, cancellable: true }, async (progress, cancellationToken) => {
-        const message: string = context.loadingMessage || localize('loadingAll', 'Loading resources...');
-        const messageDelay: number = context.loadingMessageDelay !== undefined ? context.loadingMessageDelay : 2;
-        const timer: NodeJS.Timer = setTimeout(() => progress.report({ message }), messageDelay * 1000);
+    if (notificationActive) {
+        return await callback({
```

Comment:
You can still use the below `window.withProgress` instead of creating your own cancellation token here. The only thing you need to prevent from happening with `notificationActive` is `progress.report`

Reply:
Actually scrap that. The cancellationToken you pass here should be the same cancellation token from the original callback. Since there's only one notification, I assume we want to cancel both callbacks if the user hits the "Cancel" button?

Reply:
I may of misunderstood what you were suggesting here. Would this involve caching the cancellation token from the first `runWithLoadingNotification` call to use for subsequent calls?

Reply:
Yes

Looks like you changed to match my first comment in this thread, which should be completely ignored in favor of the second comment